app/lib/shared/urls.dart

```diff
@@ -116,3 +118,52 @@ String dartSdkMainUrl(String version) {
   final url = p.join(_apiDartlangOrg, channel, version);
   return '$url/';
 }
+
+String inferRepositoryUrl(String homepage) {
+  if (homepage == null) {
+    return null;
+  }
+  final uri = Uri.tryParse(homepage);
+  if (uri == null) {
+    return null;
+  }
+  if (uri.scheme != 'http' && uri.scheme != 'https') {
+    return null;
+  }
+  if (uri.host == 'github.com' || uri.host == 'gitlab.com') {
```

Comment:
What if the host has a port set? Then we shouldn't do this, or only if 80 or 443?

Reply:
None of the git hosting services provide alternative ports, so I think we won't do much harm if we just ignore it here.

Reply:
I've added a check for non-standard url - if it has a port set, we don't infer the repo and issue tracker url. I'd rather not overdo this detection/normalization part.